### PR TITLE
combine CPU and wallclock tables in TimerOutput

### PR DESCRIPTION
**Pull request by davydden**
~~based on https://github.com/dealii/dealii/pull/6692 and as such shall be merged after.~~

I noticed that `cpu_and_wall_times` prints two tables. IMO that's waste of space and is not convenient. Therefore I combine them into the single table. Changes to `tests/base/timer_08.output` demonstrate the effect.

We can of course discuss how the header of the combined table should look and I can try adjusting the current code.

### CODE REVIEW

**Comment by drwells** [comment]
I actually like having them separated. Would you be willing make this style optional, i.e., by adding a boolean flag somewhere?

**Comment by drwells** [comment]
I have a patch buried somewhere where I refactored out this printing routine as a lambda. I cannot recall if I actually finished it, but I do remember that there was a lot of copying and pasting here.

I consider them distinct entities and would prefer to view them in two different tables. I do not have any strong arguments for why this is better or worse, its just my preference. I also have some reservations about changing whats printed to the screen in a backwards-incompatible way.

**Comment by jppelteret** [comment]
I suppose I agree with @drwells - what the desired output should be is a matter of taste and utility. Given that this is the way the output has been for a long time, it does therefore make sense to make this an option rather than force it on everyone. If one parses the outputs using a script (like I do), then the new format would mean refactoring such post-processing scripts.

**Comment by davydden** [comment]
ok, since it does not really matter much, I rather close this PR to avoid huge code duplication that would be needed to support both formats.

**Comment by davydden** [comment]
@drwells @jppelteret on a second though, I will re-work this PR so that grouping is optional...

**Comment by drwells** [comment]
In principle we should be able to simply dump everything in a `TableHandler` and make that print everything for us, but that may be more trouble than its worth.

**Comment by davydden** [comment]
> In principle we should be able to simply dump everything in a TableHandler and make that print everything for us, but that may be more trouble than its worth.

It's just a 30 seconds ago @jppelteret was in my office and we were looking at exactly the same thing 😄

**Comment by davydden** [comment]
> but that may be more trouble than its worth.

It seems the TimerOutput table is too fancy for `TableHandler`. [org_mode_table](https://www.dealii.org/developer/doxygen/deal.II/group__textoutput.html#gga84255fd53f3c75dcbdc4706079d56d8aab8dfaad92bc515243a8a3704f92f6a1a) would probably be the closest, which would result in something like
```
| Section                         | no  calls |  CPU time  | % of total |
| Hello? Hello? Hello?            |           |          s |          % |
| Is there anybody in there?      |           |          s |          % |
```
What is missing is:

* horizontal dividers before/after column labels. Even straight `-` would be ok, as doing it properly with super-columns is a PITA.
* some place to put total time (CPU/ wallclock).  One can probably work it out with super-column (i.e. create a super column with title `, maybe like

```
|                                             |  Total 825s             |
| Section                         | no  calls |  CPU time  | % of total |
```

File: doc/news/changes/minor/20180529DenisDavydov_b [hunk 1]
```diff
@@ -0,0 +1,4 @@
+New: TimerOutput::cpu_and_wall_times_grouped will print CPU and
+wallclock time in the a single table.
```
Comment (masterleinad) [inline doc/news/changes/minor/20180529DenisDavydov_b]: ~~the~~

File: source/base/timer.cc [hunk 2]
```diff
@@ -577,26 +749,66 @@ TimerOutput::print_summary() const
...
       // generate a nice table
-      out_stream << "\n\n"
-                 << "+---------------------------------------------"
-                 << extra_dash << "+------------"
-                 << "+------------+\n"
-                 << "| Total CPU time elapsed since start          "
+      out_stream << "\n\n+---------------------------------------------"
+                 << extra_dash << "+";
+
+      if (output_type != wall_times)
```
Comment (masterleinad) [inline source/base/timer.cc]: Isn't this the branch for `output_type == cpu_and_wall_times_grouped`?

File: source/base/timer.cc [hunk 3]
```diff
@@ -558,15 +558,187 @@ TimerOutput::print_summary() const
...
+      double       total_cpu_time =
         Utilities::MPI::sum(timer_all(), mpi_communicator);
 
       // check that the sum of all times is less or equal than the total time.
       // otherwise, we might have generated a lot of overhead in this function.
       double check_time = 0.;
+
       for (std::map<std::string, Section>::const_iterator i = sections.begin();
            i != sections.end();
            ++i)
```
Comment (masterleinad) [inline source/base/timer.cc]: We could use a range-based loop here.

**Comment by davydden** [comment]
@masterleinad I updated the PR, please have another look.

File: source/base/timer.cc [hunk 4]
```diff
@@ -750,64 +751,27 @@ TimerOutput::print_summary() const
...
-
-      out_stream << std::endl;
+                 << extra_dash << "+"
+                 << "------------+------------+"
+                 << "------------+------------+"
+                 << "\n"
+                 << "| Total CPU/wall time elapsed since start     "
+                 << extra_space << "|" << std::setw(10) << std::setprecision(3)
+                 << std::right << total_cpu_time << "s |            |"
+                 << std::setw(10) << std::setprecision(3) << std::right
```
Comment (masterleinad) [inline source/base/timer.cc]: It should actually not be necessary to repeat this line.

File: source/base/timer.cc [hunk 5]
```diff
@@ -750,64 +751,27 @@ TimerOutput::print_summary() const
...
+                 << total_wall_time << "s |            |"
+                 << "\n|                                             "
+                 << extra_space << "|"
+                 << "            |            |"
+                 << "            |            |"
+                 << "\n| Section                         " << extra_space
+                 << "| no. calls |" << std::setw(10) << std::setprecision(3)
```
Comment (masterleinad) [inline source/base/timer.cc]: Again, these settings have been set a few lines above.

**Pull request merged**